envoy/formatter/http_formatter_context.h

```diff
@@ -17,6 +17,16 @@ using AccessLogType = envoy::data::accesslog::v3::AccessLogType;
  */
 class HttpFormatterContext {
 public:
+  /**
+   * Interface for a context extension which can be used to provide non-HTTP specific data to
+   * formatters. This could be used for non-HTTP protocols to provide protocol specific data to
+   * formatters.
+   */
+  class Extension {
```

Comment:
This interface seems to be an opaque one (no methods).
Will it be possible to add specific methods to the interface. Having an open-ended interface isn't good in this case IMHO, and will make both the implementers and users life somewhat challenging.

Please also consider renaming this from Extension to a more specific name (e.g., CustomFormatterContext).

Reply:
> Will it be possible to add specific methods to the interface. Having an open-ended interface isn't good in this case IMHO, and will make both the implementers and users life somewhat challenging.

This `Extension` is more like the `FilterState::Object` and it may have complete different implementations for different protocols. The custom substitution formatter commands may convert it to specific data with dynamic_cast.

> Please also consider renaming this from Extension to a more specific name (e.g., CustomFormatterContext).

I have no strong point on this. But I think this is extension on the original context (`HttpFormatterContext`) rather than a new context?

Reply:
I created a new PR to split the core code change out. https://github.com/envoyproxy/envoy/pull/38232

Reply:
I've replied on the new PR. Let's continue the discussion there.
/wait-any